plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftSparkMetastoreUtil.java

```diff
@@ -0,0 +1,166 @@
+import io.trino.plugin.hive.metastore.HiveColumnStatistics;
+import io.trino.plugin.hive.type.PrimitiveTypeInfo;
+import io.trino.plugin.hive.type.TypeInfo;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createBinaryColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createBooleanColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDateColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDecimalColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDoubleColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createIntegerColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createStringColumnStatistics;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreParameterParserUtils.toDate;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreParameterParserUtils.toDecimal;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreParameterParserUtils.toDouble;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreParameterParserUtils.toLong;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.NUM_FILES;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.NUM_ROWS;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.RAW_DATA_SIZE;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.TOTAL_SIZE;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getTotalSizeInBytes;
+import static io.trino.plugin.hive.type.Category.PRIMITIVE;
+
+final class ThriftSparkMetastoreUtil
+{
+    private static final String SPARK_SQL_STATS_PREFIX = "spark.sql.statistics.";
+    private static final String COLUMN_STATS_PREFIX = SPARK_SQL_STATS_PREFIX + "colStats.";
+    private static final String COLUMN_MIN = "min";
+    private static final String COLUMN_MAX = "max";
+
+    private ThriftSparkMetastoreUtil() {}
+
+    public static boolean hasSparkColumnStatistics(Map<String, String> parameters)
+    {
+        return parameters.entrySet().stream()
+                .anyMatch(entry -> entry.getKey().startsWith(COLUMN_STATS_PREFIX));
+    }
+
+    private static boolean hasSparkStatistics(Map<String, String> parameters)
+    {
+        return parameters.entrySet().stream()
+                .anyMatch(entry -> entry.getKey().startsWith(SPARK_SQL_STATS_PREFIX));
+    }
+
+    public static Optional<HiveBasicStatistics> getHiveBasicStatistics(Map<String, String> parameters)
+    {
+        if (!hasSparkStatistics(parameters)) {
+            return Optional.empty();
+        }
+        OptionalLong numFiles = toLong(parameters.get(SPARK_SQL_STATS_PREFIX + NUM_FILES));
```

Comment:
numFiles -> fileCount
numRows -> rowCount

Reply:
same for io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil#getHiveBasicStatistics(java.util.Map<java.lang.String,java.lang.String>, boolean)